### PR TITLE
Removal of shaders! macro

### LINKED ISSUE #552: Fix or refactor shader! macro
Apparently, it's causing build issues now, according to https://github.com/PistonDevelopers/hematite/pull/193

As a suggestion, we could remove the macro entirely. Instead, one can just write this instead:
```rust
gfx::ShaderSource {
    glsl_120: Some(<shader_code>),
    .. Default::default(),
}
```
This is a bit wordier but has less magic in it. We've got to revamp all the shader ergonomics anyway.

### PR DESCRIPTION
**Pull request by kvark**
Closes #552

### CODE REVIEW

File: examples/terrain/main.rs [hunk 1]
```diff
@@ -151,10 +118,6 @@ fn main() {
     let mut glfw = glfw::init(glfw::FAIL_ON_ERRORS)
         .ok().expect("Failed to initialize glfs-rs");
 
-    glfw.window_hint(glfw::WindowHint::ContextVersion(3, 2));
```
Comment (csherratt) [inline examples/terrain/main.rs]: @bjz I recall there was at least one platform that behaved strangely if you did not provide these. Do you remember?

**Comment by csherratt** [comment]
Even if the window_hint removal was wrong, we can easily add it back in as a regression. It only effects the examples. I am going to merge this.

File: examples/terrain/main.rs (hunk 1 above)
Comment (bjz) [inline examples/terrain/main.rs]: OS X needs the window hints.

**Comment by kvark** [comment]
Thanks guys!

I was thinking, the context issues will be largely addressed by #529, but then noticed #555.

**Pull request merged**